README.md

```diff
@@ -310,6 +325,54 @@ function assignId (req, res, next) {
 }
 ```
 
+#### arguments to custom token formats
+
+Example of a custom token format that uses an argument. It will output the
+header related to the argument passed to the token.
+
+```js
+var express = require('express')
+var morgan = require('morgan')
+
+morgan.token('header', function getId (req, res, arg) {
+  return String(req.headers[arg])
+})
+
+var app = express()
+
+app.use(morgan(':header[host] :response-time'))
+
+app.get('/', function (req, res) {
+  res.send('hello, world!')
+})
+```
+
+### use custom format function
+
+Sample app that will use a custom formatting function. This will print JSON
+instead of a simple string.
+
+```js
+var express = require('express')
+var morgan = require('morgan')
+
+var formatter = morgan.compile('[:date[clf]] :method :url :res[content-length]')
+var JSONOutput = function (morgan, req, res) {
```

Comment:
Especially an issue since it's confusing when you're shadowing a variable (there is already a `var morgan = ` above, making it hard for users to sometimes understand what is actually being used here--the global or the argument).